### ISSUE TITLE
--wp-inline flag not catching inlined function calls between `.so` files 

### ISSUE THREAD

**Issue by DieracDelta** [draperlaboratory/cbat_tools#132]
I've written two `.so` files and I'm trying to use CBAT to catch the differences in `example`. `main_1` looks like:
```
#include <stdint.h>
#include <stdlib.h>
int8_t init(){
        return 0;
}

int8_t example(){
        int8_t err = init();
        return err;
}
```

`main_2` looks like:
```
#include <stdint.h>
#include <stdlib.h>

int8_t init(){
        return 1;
}

int8_t example(){
        int8_t err = init();
        return err;
}
```
The expected result when diffing these `example` in these two libraries is SAT  because `init()` returns a different value in each library. I compile and run with the following script:
```
run() {
        gcc -fPIC -fno-stack-protector -shared -Wall -o main_1.so main_1.c
        gcc -fPIC -fno-stack-protector -shared -Wall -o main_2.so main_2.c
        bap main_1.so --pass=save-project --save-project-filename=main_1.bpj
        bap main_2.so --pass=save-project --save-project-filename=main_2.bpj
        bap /bin/echo --pass=wp --wp-compare=true --wp-file1=main_1.bpj --wp-file2=main_2.bpj --wp-function=example --wp-inline="init"
}

clean() {
        rm main_1.so main_2.so main_1.bpj main_2.bpj
}

clean ; run
```
This returns UNSAT. The dot files for `example` in `main_1` and `main_2` are identical and the following:
```
000005b7: program
000005a0: sub example()
000003b8: 
000003bf: #44 := RBP
000003c2: RSP := RSP - 8
000003c5: mem := mem with [RSP, el]:u64 <- #44
000003cc: RBP := RSP
000003da: #45 := RSP
000003dd: RSP := RSP - 0x10
000003e0: CF := #45 < 0x10
000003e3: OF := high:1[(#45 ^ 0x10) & (#45 ^ RSP)]
000003e6: AF := 0x10 = (0x10 & (RSP ^ #45 ^ 0x10))
000003e9: PF := ~low:1[let $1 = RSP >> 4 ^ RSP in let $2 = $1 >> 2 ^ $1 in
$2 >> 1 ^ $2]
000003ec: SF := high:1[RSP]
000003ef: ZF := 0 = RSP
000003f6: RAX := 0
000003ff: RSP := RSP - 8
00000402: mem := mem with [RSP, el]:u64 <- 0x6AD
00000405: call @init with return %00000407

00000407: 
0000040c: mem := mem with [RBP - 1] <- low:8[RAX]
00000413: RAX := pad:64[pad:32[mem[RBP - 1]]]
0000041c: RSP := RBP
0000041f: RBP := mem[RSP, el]:u64
00000422: RSP := RSP + 8
0000042b: #48 := mem[RSP, el]:u64
0000042e: RSP := RSP + 8
00000432: call #48 with noreturn
```

The dot file for `init` in `main_1` is as follows:
```
000005b7: program
000005b4: sub init()
00000380: 
00000387: #42 := RBP
0000038a: RSP := RSP - 8
0000038d: mem := mem with [RSP, el]:u64 <- #42
00000394: RBP := RSP
0000039b: RAX := 0
000003a3: RBP := mem[RSP, el]:u64
000003a6: RSP := RSP + 8
000003af: #43 := mem[RSP, el]:u64
000003b2: RSP := RSP + 8
000003b6: call #43 with noreturn
```
And the dot files for `init` in `main_2` is:
```
000005b7: program
000005b4: sub init()
00000380: 
00000387: #42 := RBP
0000038a: RSP := RSP - 8
0000038d: mem := mem with [RSP, el]:u64 <- #42
00000394: RBP := RSP
0000039b: RAX := 1
000003a3: RBP := mem[RSP, el]:u64
000003a6: RSP := RSP + 8
000003af: #43 := mem[RSP, el]:u64
000003b2: RSP := RSP + 8
000003b6: call #43 with noreturn
```
The difference here is the return value (as expected). It is worth noting that this works as expected when both files are compiled with a main method to an executable binary with no `fPIC` and `shared` flags. In the latter case, I get the following countermodel (as expected), stating that the return values are not equal:
```
Refuted goals:
RAX0 = RAX022:
```
Similarly, if I compare object files compiled without `-fPIC`, but with `-shared` and `-c` for a static library, CBAT also produces SAT a similar countermodel:
```
Refuted goals:
RAX0 = RAX010:
```

**Comment by codyroux** [comment]
Thanks for posting this, @DieracDelta! We're slowly working towards it.

**Comment by fortunac** [comment]
This has been fixed in BAP 2.1.0-alpha and the test has been added in #145.
